test/functional/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -370,4 +372,47 @@ public void test_boostrap_return_constant_MethodType() {
 		AssertJUnit.assertEquals(expected, mt);
 		AssertJUnit.assertTrue(expected == mt);
 	}
+
+	// test that if resolved CallSite is null, the same error is rethrown
+	@Test(groups = { "level.extended" })
+	public void test_CallSiteNullErrorRethrown () {
+		/* The bootstrap method associated with the indy call in test_CallSiteNullErrorRethrown
+		 * will return null the first time its called, and a valid CallSite for all repeat calls.
+		 */
+
+		// Java 8: NullPointerException is expected on the first run
+		// Java 11: BootstrapMethodError is expected on the first run
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+			Assert.fail("BootstrapMethodError or NullPointerException should be thrown.");
+		} catch ( java.lang.Throwable e) {
+			if (VersionCheck.major() >= 11) {
+				if ( false == (e instanceof java.lang.BootstrapMethodError ) ) {
+					Assert.fail("Expected a BootstrapMethodError to be thrown");
+				}
+			} else {
+				/* Versions below Java 11 expect a NullPointerException */
+				if ( false == (e instanceof java.lang.NullPointerException ) ) {
+					Assert.fail("Expected a NullPointerException to be thrown");
+				}
+			}
+		}
+
+		// Java 8: CallSite resolution is expected to succeed
+		// Java 11 :The same BSME is expected on the second run
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+			/* success for Java 8 */
+			if ( VersionCheck.major() >= 11 ) {
+				Assert.fail("Expected a BootstrapMethodError to be thrown");
+			}
+		} catch ( java.lang.BootstrapMethodError e ) {
+			/* success for Java 11+ */
+			if ( VersionCheck.major() < 11) {
+				Assert.fail("Before Java 11 the error for a null CallSite is not stored.");
+			}			
+		} catch ( java.lang.Throwable e) {
+			Assert.fail("Unexpected throwable");
```

Comment:
This isn't required - just let the test throw and the test framework will handle it